src/pfs/fuse/fuse.go

```diff
@@ -0,0 +1,127 @@
+package fuse
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+
+	"bazil.org/fuse"
+	"bazil.org/fuse/fs"
+	"github.com/pachyderm/pachyderm/src/pfs"
+	"github.com/pachyderm/pachyderm/src/pfs/pfsutil"
+	"golang.org/x/net/context"
+)
+
+func Mount(apiClient pfs.ApiClient, repositoryName string, commitID string, mountPoint string) error {
+	if err := os.MkdirAll(mountPoint, 0777); err != nil {
+		return err
+	}
+	c, err := fuse.Mount(
+		mountPoint,
+		fuse.FSName("pfs"),
+		fuse.Subtype("pfs"),
+		fuse.VolumeName("pfs://"+repositoryName),
+	)
+	if err != nil {
+		return err
+	}
+	defer c.Close()
+
+	if err := fs.Serve(c, &filesystem{apiClient, repositoryName, commitID}); err != nil {
+		return err
+	}
+
+	// check if the mount process has an error to report
+	<-c.Ready
+	return c.MountError
+}
+
+func Unmount(mountPoint string) error {
+	return nil
+}
+
+type filesystem struct {
+	apiClient      pfs.ApiClient
+	repositoryName string
+	commitID       string
+}
+
+func (f *filesystem) Root() (fs.Node, error) {
+	return &directory{f, "/"}, nil
+}
+
+type directory struct {
+	fs   *filesystem
+	path string
+}
+
+func (*directory) Attr(ctx context.Context, a *fuse.Attr) error {
+	a.Inode = 1
+	a.Mode = os.ModeDir | 0555
+	return nil
+}
+
+func nodeFromFileInfo(fs *filesystem, fileInfo *pfs.FileInfo) (fs.Node, error) {
+	switch fileInfo.FileType {
+	case pfs.FileType_FILE_TYPE_NONE:
```

Comment:
NONE types in the protos should be considered an error case = you have to always start enums with a 0 value, and a 0 value in src/pfs means that it was not set - if this field is expected to be set, then `pfs.FieType_FILE_TYPE_NONE` should fall through to the default statement